### PR TITLE
Suppress string interpolation warnings

### PR DESCRIPTION
**Pull request by e78l**
Suppresses new warnings from https://github.com/apple/swift/pull/5195

- Avoid optional where no impact w/IUO
- Mostly cast to Optional
- Change in Notification description() to avoid issue

### CODE REVIEW

**Comment by parkera** [comment]
Why do we have to cast things to optional all over the place in the tests?

**Comment by CodaFi** [comment]
We considered three ways to silence this warning:

1) Explicitly request the debug description of the optional to be printed with `String(describing:)`
2) Unwrap with a `guard` or `if let` statement (or `!`) - we recommend `?? <default>` in the fixit
3) Cast to `Any` or `Optional` to make it explicit

I think some of these could better be handled by providing defaults than by force unwrapping or casting.

File: Foundation/NSKeyedArchiver.swift [hunk 1]
```diff
@@ -244,7 +244,7 @@ open class NSKeyedArchiver : NSCoder {
         if objv != nil &&
             self.requiresSecureCoding &&
             !NSKeyedArchiver._supportsSecureCoding(objv) {
-            fatalError("Secure coding required when encoding \(objv)")
```
Comment (CodaFi) [inline Foundation/NSKeyedArchiver.swift]: You should change this to bind `obj` in an `if-let` and remove the force unwrap.

File: Foundation/Notification.swift [hunk 2]
```diff
@@ -40,9 +40,12 @@ public struct Notification : ReferenceConvertible, Equatable, Hashable {
...
         return name.rawValue.hash
     }
-    
+
     public var description: String {
-        return "name = \(name.rawValue),  object = \(object), userInfo = \(userInfo)"
+        var description = "name = \(name.rawValue)"
+        if let obj = object { description += ", object = \(obj)" }
+        if let info = userInfo { description += ", userInfo = \(info)" }
+        return description
```
Comment (CodaFi) [inline Foundation/Notification.swift]: This changes the behavior of `description`.  Is it relevant to see `nil` in here?

**Comment by e78l** [comment]
Hello Tony and Robert,
Thanks for taking a look!
In general, I was trying to change as little as necessary –
For a string like: "expected result was \(value)", I felt it briefest to cast the value to Optional: "expected result was \(value as Optional)"
The only real change, I think, is in Notification which (as Robert points out) may affect compatibility - should the code in my comment be used instead?

**Comment by parkera** [comment]
I'm not too concerned about maintaining an exact match of the description method on `Notification` - as you point out, it's supposed to be for debugging only, anyway.

**Comment by CodaFi** [comment]
Well then, LGTM unless there's a convention you want to enforce around optional values in interpolation segments.

**Comment by CodaFi** [comment]
@swift-ci please test.

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**